iceoryx_posh/include/iceoryx_posh/popo/active_call_set.hpp

```diff
@@ -0,0 +1,212 @@
+    void detachEvent(T& eventOrigin) noexcept;
+
+    /// @brief Returns the capacity of the ActiveCallSet
+    /// @return capacity of the ActiveCallSet
+    static constexpr uint64_t capacity() noexcept;
+
+    /// @brief Returns the size of the ActiveCallSet
+    /// @return size of the ActiveCallSet
+    uint64_t size() const noexcept;
+
+  protected:
+    ActiveCallSet(EventVariableData* eventVariable) noexcept;
+
+  private:
+    class Event_t;
+
+    void threadLoop() noexcept;
+    cxx::expected<uint32_t, ActiveCallSetError>
+    addEvent(void* const origin,
+             const uint64_t eventType,
+             const uint64_t eventTypeHash,
+             CallbackRef_t<void> callback,
+             TranslationCallbackRef_t translationCallback,
+             const cxx::MethodCallback<void, uint64_t> invalidationCallback) noexcept;
+
+    void removeTrigger(const uint64_t index) noexcept;
+
+  private:
+    enum class NoEnumUsed
+    {
+        PLACEHOLDER = 0
+    };
+
+    class Event_t
+    {
+      public:
+        ~Event_t();
+
+        bool isEqualTo(const void* const origin, const uint64_t eventType, const uint64_t eventTypeHash) const noexcept;
+        bool reset() noexcept;
+        void init(const uint64_t eventId,
+                  void* const origin,
+                  const uint64_t eventType,
+                  const uint64_t eventTypeHash,
+                  CallbackRef_t<void> callback,
+                  TranslationCallbackRef_t translationCallback,
+                  const cxx::MethodCallback<void, uint64_t> invalidationCallback) noexcept;
+        void executeCallback() noexcept;
+        bool isInitialized() const noexcept;
+
+      private:
+        static constexpr uint64_t INVALID_ID = std::numeric_limits<uint64_t>::max();
+
+        void* m_origin = nullptr;
+        uint64_t m_eventType = INVALID_ID;
+        uint64_t m_eventTypeHash = INVALID_ID;
+
+        CallbackPtr_t<void> m_callback = nullptr;
+        TranslationCallbackPtr_t m_translationCallback = nullptr;
+
+        uint64_t m_eventId = INVALID_ID;
+        cxx::MethodCallback<void, uint64_t> m_invalidationCallback;
+    };
+
+    class IndexManager_t
+    {
+      public:
+        IndexManager_t() noexcept;
+        bool pop(uint32_t& index) noexcept;
+        void push(const uint32_t index) noexcept;
+        uint64_t indicesInUse() const noexcept;
+
+        uint32_t m_loffliStorage[concurrent::LoFFLi::requiredMemorySize(MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET)
+                                 / sizeof(uint32_t)];
+        concurrent::LoFFLi m_loffli;
+        std::atomic<uint64_t> m_indicesInUse{0U};
+    } m_indexManager;
+
+
+    std::thread m_thread;
+    concurrent::smart_lock<Event_t, std::recursive_mutex> m_events[MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET];
+    std::mutex m_addEventMutex;
+
+    std::atomic_bool m_wasDtorCalled{false};
+    EventVariableData* m_eventVariable = nullptr;
+    EventListener m_eventListener;
```

Comment:
Please initalize `m_eventListener` with `nullptr`. With the next refactoring the c'tor could be changed in such a way that you can construct a `ActiveCallSet` without initalizing this member.

Reply:
The eventListener requires a reference to an `EventVariableData` therefore I have to acquire a valid `EventVariableData` object